src/shared.c

```diff
@@ -1421,3 +1473,31 @@ color_t ColorFromHSV(const vec3_t hsv) {
 	ColorFromVec3(color_float, &out);
 	return out;
 }
+
+/**
+ * @brief Case-insensitive version of g_str_equal
+ */
+gboolean
+g_stri_equal (gconstpointer v1,
+             gconstpointer v2)
+{
+  const gchar *string1 = v1;
+  const gchar *string2 = v2;
+
+  return stricmp(string1, string2) == 0;
+}
+
+/**
+ * @brief Case-insensitive version of g_str_hash
+ */
+guint
+g_stri_hash (gconstpointer v)
+{
+  const signed char *p;
+  guint32 h = 5381;
+
+  for (p = v; *p != '\0'; p++)
+    h = (h << 5) + h + tolower(*p);
+
+  return h;
+}
```

Comment:
These were copied directly from glib so I didn't change their actual code other than tolower, but I can do that.